apps/factory/src/server/api/routers/likes.ts

```diff
@@ -0,0 +1,110 @@
+import { createTRPCRouter, protectedProcedure, publicProcedure } from "../trpc";
+import { createLikeInput, getLikeInput } from "~/validators/like";
+
+export const likeRouter = createTRPCRouter({
+  createLike: protectedProcedure
+    .input(createLikeInput)
+    .mutation(async ({ input, ctx }) => {
+      const { EntityId, EntityType, hasLiked, LikeId } = input;
+
+      return ctx.prisma
+        .$transaction(async (prisma) => {
+          if (hasLiked) {
+            // Delete the existing like from LikeUser table
+            await prisma.likeUser.deleteMany({
+              where: {
+                userId: ctx.session?.user.id,
+                likeId: LikeId,
+              },
+            });
+
```

Comment:
Create seperate action for unlike and like, don't use the same mutation. We need to enable caching and other mechanism for the same in the near future.